samples/bluetooth/mesh/sensor_server/src/model_handler.c

```diff
@@ -321,11 +325,177 @@ static struct bt_mesh_sensor time_since_presence_detected = {
 	.get = time_since_presence_detected_get,
 };
 
+static void amb_light_level_gain_get(struct bt_mesh_sensor_srv *srv,
+				     struct bt_mesh_sensor *sensor,
+				     const struct bt_mesh_sensor_setting *setting,
+				     struct bt_mesh_msg_ctx *ctx,
+				     struct sensor_value *rsp)
+{
+	(void)sensor_value_from_double(rsp, amb_light_level_gain);
+	printk("Ambient light level gain: %f\n", amb_light_level_gain);
+};
+
+static void amb_light_level_gain_store(double gain)
+{
+	amb_light_level_gain = gain;
+
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		int err;
+
+		err = settings_save_one("amb_light_level/gain",
```

Comment:
Sensor setting meta string is much bigger than value itself. Suggesting to optimize flash footprint.
```suggestion
		err = settings_save_one("alsg",
```

Reply:
It has a better explanation what is stored.